options.go

```diff
@@ -0,0 +1,27 @@
+package manifestival
+
+import "github.com/go-logr/logr"
+
+type options struct {
+	recursive bool
+	logger    logr.Logger
+	client    Client
+}
+
+type option func(*options)
+
+func Recursive(opts *options) {
+	opts.recursive = true
+}
+
+func UseLogger(log logr.Logger) option {
```

Comment:
Doesn't this warn about `option` not being a public type but being returned from a public function or something like that? `option` should probably be public in this case.

Reply:
Good eye! I haven't actually tested it outside the package yet, so I'll confirm there's a warning and fix.

Reply:
The warning is probably from one of the main linters, prolly `go vet` or `golint`.